sql/src/main/java/io/crate/analyze/repositories/RepositoryParamValidator.java

```diff
@@ -51,7 +52,9 @@ public Settings convertAndValidate(String type, Optional<GenericProperties> gene
             throw new IllegalArgumentException(String.format(Locale.ENGLISH, "Invalid repository type \"%s\"", type));
         }
 
-        genericProperties = typeSettings.preProcess(genericProperties);
+        Optional<GenericProperties> dynamicProperties = typeSettings.dynamicProperties(genericProperties);
+        genericProperties = typeSettings.removeProperties(genericProperties, dynamicProperties);
```

Comment:
instead of removing them and therefore rebuilt the generic properties, would it make sense to just exclude the dynamicProperties from the invalidSettings Set further down?